### DISCUSSION TITLE
Key.getExpirationTime() now doesn't have parameter "capability".

### DISCUSSION THREAD

**Discussion by IvanPizhenko** [openpgpjs/openpgpjs#1493] [Q&A]
In v4, it was:
```typescript
class Key {
...
      getExpirationTime(capability?: 'encrypt' | 'encrypt_sign' | 'sign' | null, keyId?: Keyid | null, userId?: UserId | null): Promise<Date | typeof Infinity | null>; // Returns null if `capabilities` is passed and the key does not have the specified capabilities or is revoked or invalid.
...
}
```

In v5, there is no parameter `capability`:
```typescript
export class Key {
...
  public getExpirationTime(userID?: UserID, config?: Config): Promise<Date | typeof Infinity | null>;
...
}
```
What is instead of it?
With v4, we are calling it as follows: `key.getExpirationTime('encrypt')`. What should we do with v5 to have an equivalent?
@larabr - any comments?

**Comment by larabr** [comment]
There is no equivalent in v5.
To get the expiration time of a specific subkey (e.g. the one returned by [`Key.getEncryptionKey()`](https://github.com/larabr/openpgpjs/blob/master/src/key/key.js#L288), you can call `subkey.getExpirationTime()`. However, if the primary key expires before the subkey, the subkey will also be considered invalid, so you should effectively take the minimum between the primary key expiration (which you get with `Key.getExpirationTime()`) and the subkey expiration time.

To get the latest expiration time across all encryption subkeys, which is similar (but not equivalent) to what v4 did, you can call [`Key.getDecryptionKeys(null, null)`](getDecryptionKeys) to get all encryption subkeys, and take the maximum of their expiration times, and then again compare it to the primary key expiration.
However, if you do this, note that you will not detect whether there are gaps in the validity of encryption subkeys: even though the latest key expiration is at some time T, it could be that no encryption key is valid at some point before T (e.g. because a key expires before another key starts being valid).
This type of edge case is part of the reason why we dropped the `capabilities` argument and overall simplified (& clarified) the behaviour of `Key.getExpirationTime()`.
(Also, `getDecryptionKeys` is only available for private keys, but maybe you can copy the code and create a helper function that does the same for any `Key` object).
